### ISSUE TITLE
BatArray.remove_at

### ISSUE THREAD

**Issue by UnixJunkie** [ocaml-batteries-team/batteries-included#995]
Would be the reverse of the existing:
```
val BatArray.insert: 'a array -> 'a -> int -> 'a array
```

**Comment by UnixJunkie** [comment]
I propose:
```
val BatArray.remove: 'a array -> int -> 'a * 'a array
```

**Comment by UnixJunkie** [comment]
I had to code it yesterday...
It annoys me when I encounter small functionalities that batteries doesn't have.

**Comment by fccm** [comment]
has been merged with PR #996
Thanks!
